src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -510,28 +542,54 @@ private ODataNestedResourceInfo ReadNestedResourceInfo()
             {
                 if (feed != null)
                 {
-                    MaterializerNavigationLink.CreateLink(link, feed, this.materializerContext);
+                    nestedEntry = MaterializerNestedEntry.CreateNestedEntry(
+                        link,
+                        new List<IMaterializerState>(),
+                        this.materializerContext);
+
+                    nestedEntry.Feed = feed;

```

Comment:
why can't we add the `materializerFeed` to the `nestedEntry`'s `nestedItems` here?

Reply:
The original `MaterializerNavigationLink` uses feed(ODataResourceSet). This allows backwards compatibility.
Unless we refactor the existing logic, which is beyond the scope of this PR